core/store/models/job_spec.go

```diff
@@ -63,11 +76,25 @@ func NewJob() JobSpec {
 // JobSpecRequest
 func NewJobFromRequest(jsr JobSpecRequest) JobSpec {
 	jobSpec := NewJob()
-	jobSpec.Initiators = jsr.Initiators
-	for _, initr := range jobSpec.Initiators {
-		initr.JobSpecID = jobSpec.ID
+	for _, initr := range jsr.Initiators {
+		jobSpec.Initiators = append(jobSpec.Initiators, Initiator{
+			JobSpecID: jobSpec.ID,
+			// Type must be downcast to comply with Initiator deserialization
+			// logic. Ideally, Initiator.Type should be its own type (InitiatorType)
+			// that handles deserialization validation.
+			Type:            strings.ToLower(initr.Type),
```

Comment:
@j16r @se3000 This comment about `InitiatorType`, and its creation. I held off from placing it in this PR, so it remains a standard `string`. This is different from say `TaskType` which was introduced later in the lifecycle of the repo. If you think I should introduce it in this PR, let me know.